spring-integration-core/src/main/resources/org/springframework/integration/config/xml/spring-integration-4.2.xsd

```diff
@@ -1659,6 +1659,91 @@
 	</xsd:complexType>
 
+	<xsd:element name="barrier">
+		<xsd:annotation>
+			<xsd:documentation>
+				Defines an endpoint that suspends a thread until a corresponding release message is received.
+				The endpoint's message handler is a 'BarrierMessageHandler'.
+			</xsd:documentation>
+		</xsd:annotation>
+		<xsd:complexType>
+			<xsd:choice minOccurs="0" maxOccurs="2">
+				<xsd:element name="request-handler-advice-chain" type="handlerAdviceChainType" minOccurs="0" maxOccurs="1" />
+				<xsd:element ref="poller" />
+			</xsd:choice>
+			<xsd:attributeGroup ref="inputOutputChannelGroup" />
+			<xsd:attribute name="id" type="xsd:string" use="required"/>
+			<xsd:attribute name="timeout" type="xsd:string" use="required">
+				<xsd:annotation>
+					<xsd:documentation>
+						The time in milliseconds to suspend the thread. See `requires-reply`.
+						Also, if the trigger message is received first, the time that thread will
+						wait before logging an error and exiting.
+					</xsd:documentation>
+				</xsd:annotation>
+			</xsd:attribute>
+			<xsd:attribute name="requires-reply" type="xsd:string" use="optional">
+				<xsd:annotation>
+					<xsd:documentation>
+						Specify whether the barrier must return a non-null value. This value will be
+						'false' by default, but if set to 'true', a ReplyRequiredException will be thrown when
+						the barrier times out.
+					</xsd:documentation>
+				</xsd:annotation>
+			</xsd:attribute>
+			<xsd:attribute name="correlation-strategy" type="xsd:string">
+				<xsd:annotation>
+					<xsd:appinfo>
+						<tool:annotation kind="ref">
+							<tool:expected-type type="java.lang.Object" />
+						</tool:annotation>
+					</xsd:appinfo>
+					<xsd:documentation>
+						A reference to a bean that implements the decision algorithm as to whether a given
+						message group is complete. The bean can be an implementation of the
+						CorrelationStrategy interface or a POJO.
+					</xsd:documentation>
+				</xsd:annotation>
+			</xsd:attribute>
+			<xsd:attribute name="correlation-strategy-method" type="xsd:string">
+				<xsd:annotation>
+					<xsd:appinfo>
+						<tool:annotation>
+							<tool:expected-method type-ref="@correlation-strategy" />
+						</tool:annotation>
+					</xsd:appinfo>
+					<xsd:documentation>
+						A method defined on the bean referenced by correlation-strategy, that implements the
+						correlation decision algorithm
+					</xsd:documentation>
+				</xsd:annotation>
+			</xsd:attribute>
+			<xsd:attribute name="correlation-strategy-expression" type="xsd:string">
+				<xsd:annotation>
+					<xsd:documentation>
+						A SpEL expression which implements correlation decision
+						algorithm to apply to the Message (e.g., payload.getPerson().getId() - correlate
+						based on the 'id' of the 'person' attribute of the message payload object)
+					</xsd:documentation>
+				</xsd:annotation>
+			</xsd:attribute>
+			<xsd:attribute name="output-processor" type="xsd:string">
```

Comment:
I thought about it, but it's not likely it is something people will want to change; and they can always wire in an `ExpressionEvaluating`...